ci/travis/ci.sh

```diff
@@ -38,21 +38,187 @@ should_run_job() {
   return "${skip}"
 }
 
+# Idempotent environment loading
 reload_env() {
   # TODO: We should really just use a new login shell instead of doing this manually.
   # Otherwise we might source a script that isn't idempotent (e.g. one that blindly prepends to PATH).
+
+  if [ -n "${GITHUB_PULL_REQUEST-}" ]; then
+    case "${GITHUB_PULL_REQUEST}" in
+      [1-9]*) TRAVIS_PULL_REQUEST="${GITHUB_PULL_REQUEST}";;
+      *) TRAVIS_PULL_REQUEST="false";;
+    esac
+    export TRAVIS_PULL_REQUEST
+  fi
+
+  local definitions=() paths=()
+
+  # Environment variables for all platforms
+  definitions+=(
+    PYTHON3_BIN_PATH=python
+    GOROOT="${HOME}/go"
+    GOPATH="${HOME}/go_dir"
+  )
+  # Environment variables for Windows
+  test "${OSTYPE}" != msys || definitions+=(
+    USE_CLANG_CL=1
+  )
+
+  # NOTE: Modifying PATH invalidates Bazel's cache! Do not add to PATH unnecessarily.
+  # PATH for all platforms
+  paths+=(
+    "${HOME}/miniconda/bin"
+  )
+  # PATH for Windows
+  test "${OSTYPE}" != msys || paths+=(
+    "${HOME}/miniconda/bin/Scripts"
+  )
+
+  export "${definitions[@]}" 2> /dev/null
```

Comment:
why do this delayed? Why not just exporting stuff directly above?

Reply:
No strict reason, I just did it out of habit since it separates out the export command (imperative) from the declarative portion. I can change it if you prefer to do it directly above.